pkg/apis/submariner.io/v1/types.go

```diff
@@ -78,8 +78,13 @@ type EndpointSpec struct {
 	NATEnabled    bool              `json:"nat_enabled"`
 	Backend       string            `json:"backend"`
 	BackendConfig map[string]string `json:"backend_config,omitempty"`
+	PortConfig    map[string]string `json:"natConfig,omitempty"`
 }
 
+const (
+	NATDiscoveryPortConfig = "nat-discovery"
```

Comment:
It could also be, Since it was in a "PortConfig" dictionary it seemed like repeating.

I want to keep the "PortConfig" as portconfig ( I need to fix the json name as you pointed out it's wrong)

The idea is that PortConfig (or simply "Ports?") would hold the port numbers also for IPSEC or wireguard, etc which now it's fixed per clusterset and doesn't really work. It needs to be flexible per-endpoint for all the on-prem nat stuff to work ok.